audio/audio_hw.c

```diff
@@ -1874,10 +1875,45 @@ static ssize_t do_in_pcm_read(struct audio_stream_in *stream, void *buffer,
         goto exit;
     }
 
-    if (in->resampler.resampler != NULL) {
-        ret = read_resampled_frames(in, buffer, frames_rq);
+    if (in->common.channel_count != in->hw_channel_count) {
+
+        ALOGV("do_in_pcm_read(): Read %d channel stream with adjustment (HW only supports %d channel reading)",
+              in->common.channel_count, in->hw_channel_count);
+
+        size_t read_buf_size = bytes * in->hw_channel_count / in->common.channel_count;
+        void *read_buffer = calloc(1, read_buf_size);
```

Comment:
It's inefficient to alloc a temporary buffer on every read. The buffer should be allocated when the stream is opened.

Reply:
Hello @rfvirgil .
I have updated the patch as you proposed.